sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateOperation.cs

```diff
@@ -60,10 +60,15 @@ public override KeyVaultCertificateWithPolicy Value
                 {
                     throw new OperationCanceledException("The operation was canceled so no value is available.");
                 }
-#pragma warning restore CA1065 // Do not raise exceptions in unexpected locations
+
+                if (Properties.Error != null)
+                {
+                    throw new InvalidOperationException("The certificate operation failed. See Properties.Error for details.");
```

Comment:
Why not include Error into the exception message itself?

Reply:
@schaabs would have to speak to that. I considered doing it in this PR, but got to thinking - and maybe this is why - there's more the error than just the message, though that's true of every exception. I can certainly add it, but the test will have to only assert the prefix since we don't control the message back from the server (at least for live/record modes).